source/fuzz/fuzzer_pass_add_function_calls.cpp

```diff
@@ -188,10 +188,6 @@ std::vector<uint32_t> FuzzerPassAddFunctionCalls::ChooseFunctionCallArguments(
     if (type_to_available_instructions.count(arg_type_id)) {
       std::vector<opt::Instruction*>& candidate_arguments =
           type_to_available_instructions.at(arg_type_id);
-      // TODO(https://github.com/KhronosGroup/SPIRV-Tools/issues/3177) The value
-      //  selected here is arbitrary.  We should consider adding this
-      //  information as a fact so that the passed parameter could be
-      //  transformed/changed.
```

Comment:
I can't remember quite what you, I and @paulthomson discussed, but I think it was that we would leave this as is (which you've done), but add an issue to favour using irrelevant zeros at first, or something like that.

Reply:
A possible solution here is to do an OpCopyObject of each selected parameter (without introducing a synonym fact), and mark the *copy* as being irrelevant.

It would be good if you could make a new issue to track what we do here and cite that issue instead of the old issue.

Reply:
I think this comment is resolved because:
* we select all available ids (irrelevant and not) as argument candidates https://github.com/KhronosGroup/SPIRV-Tools/blob/9bebdd6cacf8d7715f359eab88bdd3bc3eabd6d0/source/fuzz/fuzzer_pass_add_function_calls.cpp#L149
* if there is no suitable id in the module, we create an irrelevant zero https://github.com/KhronosGroup/SPIRV-Tools/blob/9bebdd6cacf8d7715f359eab88bdd3bc3eabd6d0/source/fuzz/fuzzer_pass_add_function_calls.cpp#L242

I think this is what we've discussed at the meeting.

Reply:
Sorry if this contradicts what we discussed, but having thought about it more and discussed with @paulthomson I think we should do the following, in the fuzzer pass:

- If the argument type is non-pointer, pass an irrelevant zero.
- If the argument type is pointer, find all "irrelevant pointee" pointers of the right type and select one at random.  If there are none, add a new "irrelevant pointee" variable and use it.

This should allow getting rid of `GetAvailableInstructionsSuitableForActualParameters`, I believe.

I don't mind whether you do this now, or open an issue for it it and address it directly after the current series of PRs for the "id is irrelevant" issue.

Let me know either way.

Reply:
Ok, I'll add fix this in the current PR.

Reply:
Fixed in 4d17c32.